packages/frontend/components/scaffold-eth/AddressInput.tsx

```diff
@@ -0,0 +1,69 @@
+import { ChangeEvent, useCallback, useEffect } from "react";
+import Blockies from "react-blockies";
+import { useEnsAddress } from "wagmi";
+
+interface IAddressInput {
+  value?: string;
+  onChange?: (arg: string) => void;
+  placeholder?: string;
+}
+// can we move this functions to other utility file ?
+const isENS = (address = "") => address.endsWith(".eth") || address.endsWith(".xyz");
+
+export const Sleep = async (time: number): Promise<any> =>
+  new Promise(resolve => setTimeout(() => resolve(true), time));
+
+/**
+ Address input component with ens name resolution
+*/
+
+const AddressInput = ({ value, onChange, placeholder }: IAddressInput): any => {
+  const { data: ensData, isLoading } = useEnsAddress({
+    name: isENS(value) ? value : "",
+  });
+
+  const onChangeAddress = async (event: ChangeEvent<HTMLInputElement>) => {
+    if (onChange) {
+      onChange(event.target.value);
+    }
+  };
+
+  const loadEnsAddress = useCallback(async () => {
+    if (ensData && onChange) {
+      await Sleep(100);
+      onChange(ensData);
+    }
+
+    if (ensData === null && onChange) {
+      // add a little delay on load
+      await Sleep(100);
+      onChange("");
+    }
+  }, [ensData, onChange]);
```

Comment:
Don't understand the need of this.

We just need the useEffect when we get the ENS resolution.